src/server.c

```diff
@@ -4291,72 +4323,75 @@ void addReplyCommand(client *c, struct redisCommand *cmd) {
             keystep = cmd->legacy_range_key_spec.fk.range.keystep;
         }
 
-        /* We are adding: command name, arg count, flags, first, last, offset, categories, additional information (map) */
-        addReplyArrayLen(c, 8);
-        addReplyBulkCString(c, cmd->name);
+        addReplyArrayLen(c, !cmd->subcommands_dict? 9: 10);
+        if (show_full_name && cmd->parent) {
+            sds fullname = sdscatfmt(sdsempty(), "%s %s", cmd->parent->name, cmd->name);
```

Comment:
maybe we should use `|` as a seperator?

Reply:
we already have `getFullCommandName`

Reply:
truth be told i'm a bit uncomfortable with this change.
i.e. that the block of response for CONFIG SET, will be different if you asked for `config` vs the one you'll get when you asked for `config|set`.
i'm not sure what would be more useful for the client.
but maybe considering that the only way to observe that is when you specifically ask for it using the `|` char, we can spit that back at the caller. so let's do that.